src/mlpack/tests/mean_shift_test.cpp

```diff
@@ -18,48 +18,54 @@
 using namespace mlpack;
 
 // Generate dataset; written transposed because it's easier to read.
-arma::mat meanShiftData("  0.0   0.0;" // Class 1.
-                        "  0.3   0.4;"
-                        "  0.1   0.0;"
-                        "  0.1   0.3;"
-                        " -0.2  -0.2;"
-                        " -0.1   0.3;"
-                        " -0.4   0.1;"
-                        "  0.2  -0.1;"
-                        "  0.3   0.0;"
-                        " -0.3  -0.3;"
-                        "  0.1  -0.1;"
-                        "  0.2  -0.3;"
-                        " -0.3   0.2;"
-                        " 10.0  10.0;" // Class 2.
-                        " 10.1   9.9;"
-                        "  9.9  10.0;"
-                        " 10.2   9.7;"
-                        " 10.2   9.8;"
-                        "  9.7  10.3;"
-                        "  9.9  10.1;"
-                        "-10.0   5.0;" // Class 3.
-                        " -9.8   5.1;"
-                        " -9.9   4.9;"
-                        "-10.0   4.9;"
-                        "-10.2   5.2;"
-                        "-10.1   5.1;"
-                        "-10.3   5.3;"
-                        "-10.0   4.8;"
-                        " -9.6   5.0;"
-                        " -9.8   5.1;");
-
+template<typename MatType>
+MatType GetMeanShiftData()
+{
+  return MatType("  0.0   0.0;" // Class 1.
+                 "  0.3   0.4;"
+                 "  0.1   0.0;"
+                 "  0.1   0.3;"
+                 " -0.2  -0.2;"
+                 " -0.1   0.3;"
+                 " -0.4   0.1;"
+                 "  0.2  -0.1;"
+                 "  0.3   0.0;"
+                 " -0.3  -0.3;"
+                 "  0.1  -0.1;"
+                 "  0.2  -0.3;"
+                 " -0.3   0.2;"
+                 " 10.0  10.0;" // Class 2.
+                 " 10.1   9.9;"
+                 "  9.9  10.0;"
+                 " 10.2   9.7;"
+                 " 10.2   9.8;"
+                 "  9.7  10.3;"
+                 "  9.9  10.1;"
+                 "-10.0   5.0;" // Class 3.
+                 " -9.8   5.1;"
+                 " -9.9   4.9;"
+                 "-10.0   4.9;"
+                 "-10.2   5.2;"
+                 "-10.1   5.1;"
+                 "-10.3   5.3;"
+                 "-10.0   4.8;"
+                 " -9.6   5.0;"
+                 " -9.8   5.1;").t();
+}
 
 /**
  * 30-point 3-class test case for Mean Shift.
  */
-TEST_CASE("MeanShiftSimpleTest", "[MeanShiftTest]")
+TEMPLATE_TEST_CASE("MeanShiftSimpleTest", "[MeanShiftTest]", float, double)
 {
+  typedef TestType ElemType;
+
   MeanShift<> meanShift;
 
   arma::Row<size_t> assignments;
-  arma::mat centroids;
-  meanShift.Cluster((arma::mat) trans(meanShiftData), assignments, centroids);
+  arma::Mat<ElemType> centroids;
+  meanShift.Cluster(GetMeanShiftData<arma::Mat<ElemType>>(), assignments,
+      centroids);
```

Comment:
No need to transpose GetMeanShiftData() ?

Reply:
The `.t()` is hidden at the end of `GetMeanShiftData()` now :)